carmm/elasticity/elastic_tensor_calculation.py

```diff
@@ -0,0 +1,186 @@
+                            stress.append(float(lines[index + 5].split()[3]))
+                            stress.append(float(lines[index + 5].split()[4]))
+                            stress.append(float(lines[index + 6].split()[2]))
+                            stress.append(float(lines[index + 6].split()[3]))
+                            stress.append(float(lines[index + 6].split()[4]))
+
+                        try:
+                            stress = np.array(stress).reshape(3,3)
+                            stress_list.append(stress)
+                            # print(stress)
+                            f.close()
+                        except Exception as e:
+                            print('Error encountered. See the message below')
+                            print(e)
+                            f.close()
+
+        stress_tensor = np.array(stress_list)
+
+        return stress_tensor
+
+    else:
+        raise ValueError(
+            f'The file extension provided is not supported. Please make sure it is one of the following [.traj, .xyz, .out]')
+
+
+def compute_elasticity_tensor(strain_tensor,stress_tensor,write_elasticity_tensor=True,write_output=True):
+    """
+    Compute the elasticity tensor from strain and stress tensors.
+
+    Parameters
+    ----------
+    strain_tensor : numpy.ndarray
+        Array of shape (N, 3, 3) containing the applied strain tensors,
+        where N is the number of deformation structures. Each element is a
+        symmetric 3×3 strain tensor.
+    stress_tensor : numpy.ndarray
+        Array of shape (N, 3, 3) containing the resulting stress tensors
+        corresponding to each strain in `strain_tensor`. Units depend on the underlying simulation code.
+    write_elasticity_tensor : bool, optional
+        If True (default), write the computed fourth-rank elasticity tensor
+        (C_{ijkl}) to disk as pickle (.pkl) file.
+    write_output : bool, optional
+        If True (default), print summary information such as the unique values in elasticity tensor,
+         complete strain and stress tensors.
+
+    Returns
+    -------
+    C : numpy.ndarray
+        The fourth-rank elasticity tensor (C_{ijkl})
+
+    Notes
+    -----
+    - The elasticity tensor is computed by solving the linear relation:
+          σ_{ij} = C_{ijkl} ε_{kl} using the Pymatgen functionality diff_fit().
+    - The function assumes small-strain elasticity.
+    - The unique values are printed in output summary because the crystal symmetry causes several
+      elements of the tensor to be equal. Hence, the unique values come in handy when comparing with literature values.
+    """
+    import pickle
+    import numpy as np
+    from pymatgen.analysis.elasticity import diff_fit
+    elasticity_tensor = diff_fit(strain_tensor, stress_tensor, order=2)
+    print(np.unique(np.array(elasticity_tensor[0]), return_index=True)[0] * 160.2716621)
+
+    if write_output:
```

Comment:
I'm not a big fan of write functions in compute functions - really compute should return an object, and write should be a separate function acting on this object. I don't mind leaving as is but just raising in case this can be easily separated.